### PR TITLE
Statemine Integration for KodaDot

### LINKED ISSUE #1452: Rename statemine routed
From /unique to /statemine

### LINKED ISSUE #1531: List of KodaDot communities
```ts
  public publicCommunity: any = [
    ['🇦🇲 Armenia', 'https://t.me/kodadotarmenia'],
    // ['🇧🇩 Bengali', 'https://t.me/KodaDot_Bengali'], seems not active
    ['🇨🇿 Česko', 'https://t.me/joinchat/Fhnvbi5a_wRjNzFk'],
    ['🇯🇵 Japan', 'https://t.me/kodadotjapan'],
    ['🇵🇱 Polska', 'https://t.me/joinchat/HG7J2RAk906N7scb'],
    ['🇵🇹 Portuguese', 'https://t.me/joinchat/1UHYFZpVYmE1OTZk'],
    ['🇷🇺 Russia', 'https://t.me/kodadotru'],
    ['🇪🇸 Spanish', 'https://t.me/joinchat/HkF3cxImJAJGoRH9'],
    ['🇹🇷 Türkiye', 'https://t.me/KodaDotTR'],
    ['🌐 Global KodaDot Discord', 'https://discord.gg/u6ymnbz4PR'],
    ['🏗 Developers', 'https://discord.gg/KkctBVQ8kP'],
    // ['🇻🇳 Việt Nam', 'https://t.me/joinchat/GR0OiagewrkTzD4u'], seems not active
    [
      'Want to help translate?',
      'https://github.com/kodadot/nft-gallery/tree/i18n/src/locales',
    ],
  ]
  ```

### LINKED ISSUE #1536: Tracking: Wen Statemine
The following list of features should be available: 
- [ ] Create Class
- [ ] Create NFT
- [ ] View Collection
- [ ] View NFT
- [ ] View Collection List
- [ ] View Profile
- [ ] Can Transfer
- [ ] Can Burn
- [ ] Can Transfer Collection
- [ ] Teleport

### LINKED ISSUE #1546: Calculate last index used
In create now I use `alreadyMinted` which is a skewed metric :)

```ts
Math.max(alreadyMinted, lastIndexUsed)
```

### PR DESCRIPTION
**Pull request by vikiival**
### Statemine is here!

- :tada: Initial commit KodaDot Statemine utils
- :zap: added pseudo-random value generator
- :top: copying routes for unique
- :top: using parametrized url from @vue-settings
- :top: added graphql queries for the statemine
- :contruction: WIP Create Pages for unique
- :zap: ability to use env variables in the nuxt config
- :construction: create collection for statemine should be back on track
- :construction: create nft for statemine should be back on track
- :bug: notification in create token shows name instead of edition
- :bug: missing <b-field> tag in create token
- :broom: moved available actions into gallery/item
- :zap: app can correctly process .env variables
- nuxt can now read graphql url from the .env
- :zap: dynamic prefixes exists works like a charm
- :zap: Gallery dynamically routes, and uses the query specially crafted for the prefix
- :zap: GalleryItem works with statemine
- :construction: WIP collection detail
- :broom: removed unused fetch pack meta
- :bug: fixed route to the collection item
- :bug: fixed grqphql queries on the profile in statemine
- :bug: fixed grqphql queries on the collection item in statemine
- :bug: fixed statemine actions
- :top: added import for custom attribute in create token
- Added english translations
- :top: added metamixin as the abstract function executor
- something
- :zap: collection view works with statemine
- :top: added prefix mixin
- :zap: profile link is automatically resolved based on prefix
- :spiderweb: we love changing types in graphql
- :bug: gallery card now shows correct name and can route to detail from collection item
- :bug: correctly checking health of the indexer
- :bug: fixed creation of collection and nfts
- :zap: upgraded to the new Gallery detail
- :zap: added delegate to show in the statemine
- :bug: working actions on statemine
- :bug: link from gallery item to the collection view is working again
- :zap: can clear metadata and attributest
- :zap: added impl for CollapseWrapper
- :top: added CollapseCardWrapper
- :bug: removed avatar because aliens
- :zap: detail can render key-value properties
- :bug: render properties only when there is at least one
- :bug: accountId should not be a required prop
- :zap: working teleport
- :construction: Danger Modal
- :zap: Abilility to transfer collection
- :bug: added accountId for transfer collection as required
- :zap: ability to clear all attributes
- :zap: added check balance for minting the collection
- :zap: can indicate that NFT is Frozen
- :broom: added translations to the customattribute input
- :bug: set default wss url for statemine
- :bug: do not preload nfts on the landing page
- :broom: comment links witch wont be functional
- :zap: onApiConnect
- :zap: faster collection item
- :bug: fixing that ugly account select
- :bug: fixed current deposit
- :top: bumped up @vue-settings
- :zap: creating new, correct types
- updated queries to fetch collection meta
- :zap: added new features to the ipfs sanitizer
- Working NFT detail for statemine
- :bug: ability to show nft list for statemine
- :shrug: I have no idea what happened
- :bug: collectionDetail works on statemine
- :bug: prevents null values to update the IndexedDB
- :bug: can handle broken collection on statemine
- filter out collections without any image
- :bug: functional routes for the statemine
- :bug: indexer has _metadata field on newer subquery
- :bug: properties were fairly broken with the bigger texts
- :globe_with_meridians: using correct default values for CustomAttributeInput
- :globe_with_meridians: showing correct translation for on-chain attributes
- :bug: fixed loader on collections
- :zap: more meaningful dev errors
- :children_crossing: banned SubstraTee because of their IPFS which timeouts on pinata
- Close #1452 renaming routes from /unique to /statemine
- :bug: fixed correct routing on the profile
- :zap: Create NFT is now showing collection name
- :top: updated vue-settings in the statemine
- :zap: new cool settings
- :bug: Button in AddOption was invisible :eyes:
- :art: can set apollo config based on the settings
- :bug: defineApolloConfig is not defined
- :construction: disabled indexer checker as I have no default
- :bug: fixed IdentityPopover for statemine
- :bug: using new client base routes
- :bug: collection can use the prefix
- :coffin: adding Eipd9P7j9XC7voYxYgAoyE6wwt8BmpLDNC89RAU2cf6Qtrk to the statemine deny list
- :bug: can show animation url on the gallery item on statemine
- :top: updated vue settings to the 0.0.38-rc.3
- :top: updated vue settings to 0.0.38-rc.5
- Navbar now uses prefixMixin
- prefixMixin now uses value from store
- :top: updated vue settings to 0.0.38-rc.6
- :zap: added prefix middleware which handles url based on prefix
- :bug: rmrk now uses prefixMixin
- :bug: collection list can properly handle loading
- :broom: moved Landing into separated component
- :zap: statemine has it's own landing
- Close #1531
- :zap: urls on landing are handled by prefix
- :zap: removed that ugly dropdown from landing
- :zap: Landing take buildOn prop
- /statemine landing is resolving statemine prefix
- :bug: isLoading is now handled correctly for gallery
- :bug: fixed client resolving on users profile
- :broom: added new placeholder
- updated unique queries to obtain metadata from collection
- :zap: GalleryCardList can decide which source will be used for metadata
- :zap: enable spotlight, series and other mints for RMRK
- :zap: Adding Landing page for RMRK
- :bug: correct naming for the GalleryPage
- ⚒️ fixing #1532 on statemine
- :bug: added Prefix mixin on Create
- :bug: NavbarProfileDropdown routes to correct prefix
- updated vue-settings
- :broom: cleared unsued properties from navbar
- :bug: collection item can navigate to correct prefix
- :bug: fixed prefix resolving for RMRK
- :zap: show admin panel only for RMRK
- :bug: removed freeze action from the statemine
- :bug: fixed bug for consuming and revoking NFT on statemine
- :bug: can correctly navigate to the gallery detail
- :bug: will set non-existent NFT as burned
- Close #1546
- :broom: cleanup old comments from create token
- Added Westmint to the statemine integration
- :soap: running linter
- :bug: added missing client on GalleryItem for RMRK
- added /detail routes for statemine and westmint

Thank you for your contribution to the KodaDot NFT gallery, 
we really appreciate your contribution!

### PR type
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

### Before submitting this PR, please make sure:
- [x] Your code builds clean without any erros or warnigns
- [x] You've posted screenshot of demonstrated change in this PR
- [x] Merged recent default branch, **main** and you have no conflicts
- [ ] Didn't break any original functionality 

### What's new? (may be part of changelog)
- Closes #1536

### CODE REVIEW

File: components/unique/Gallery/Item/DangerModal.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  protected onSuccess(block: string) {
+    showNotification(`Meta cleared in block ${block}`)
+  }
+
+  @Watch('attributes', { deep: true })
+  protected onAttributesChange(val: Attribute[], oldVal: Attribute[]) {
+
+  }
```
[the remark this answers is not in the document]
Comment (roiLeo) [reply components/unique/Gallery/Item/DangerModal.vue:85]: Good boy!

[the remark this answers is not in the document]
Comment (yangwao) [reply components/unique/Gallery/Item/DangerModal.vue:85]: [gifbot:dog]

**Comment by yangwao** [comment]
gifbot:dog

**Comment by vikiival** [comment]
[gifbot:dog]

**Comment by yangwao** [comment]
[gifbot:dog]

**Comment by yangwao** [comment]
/giphy anime spaceship coming to home base

**Comment by roiLeo** [comment]
> **A reviewer left a comment:**
> 
>     * Found a bug (first time testing this netlify feature)

HEY! That's me.

**Comment by vikiival** [comment]
> **A reviewer left a comment:**
> 
> * Found a bug (first time testing this netlify feature)

Should be fixed in 92bdea833af64d03a11b7e349fe423e478402bf3

**Comment by yangwao** [comment]
@vikiival 👀

File: pages/westmint/create.vue [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<template>
+  <div class="columns">
+    <div class="column is-6 is-offset-3">
```
Comment (roiLeo) [inline pages/westmint/create.vue:3]: missing `layout` on this file

File: components/rmrk/Gallery/History.vue [hunk 3]
```diff
@@ -41,14 +41,14 @@
...
-              <nuxt-link
+              <router-link
                 :to="{
                   name: 'profile',
                   params: { id: props.row.From },
                 }"
               >
                 <Identity :address="props.row.From" inline noOverflow />
-              </nuxt-link>
+              </router-link>
```
Comment (roiLeo) [inline components/rmrk/Gallery/History.vue:51]:
```suggestion
              </nuxt-link>
```

File: components/rmrk/Gallery/History.vue [hunk 4]
```diff
@@ -41,14 +41,14 @@
               label="From"
               v-slot="props"
             >
-              <nuxt-link
+              <router-link
```
Comment (roiLeo) [inline components/rmrk/Gallery/History.vue:44]:
```suggestion
              <nuxt-link
```

File: components/rmrk/Gallery/History.vue [hunk 5]
```diff
@@ -103,15 +98,19 @@ const components = {
...
+  From: string
+  To: string
+  Amount: string
+  Date: string
+  Block: string
+}
+
+type ChartData = {
+  buy: any[],
+  list: any[],
```
Comment (roiLeo) [inline components/rmrk/Gallery/History.vue:111]:
```suggestion
  buy: [Date, number][],
  list: [Date, number][],
```

File: components/rmrk/Gallery/Item/Name.vue [hunk 6]
```diff
@@ -21,7 +27,7 @@
 </template>
 
 <script lang="ts" >
-import { Component, Prop, Vue } from 'nuxt-property-decorator'
+import { Component, Prop, Vue } from 'vue-property-decorator'
```
Comment (roiLeo) [inline components/rmrk/Gallery/Item/Name.vue:30]:
```suggestion
import { Component, Prop, Vue } from 'nuxt-property-decorator'
```

File: components/unique/Collection/Item/TransferCollectionModal.vue [hunk 7]
```diff
@@ -0,0 +1,77 @@
+<template>
```
Comment (roiLeo) [inline components/unique/Collection/Item/TransferCollectionModal.vue:1]: - Indentation
- What does this component do?

File: components/unique/Create/CreateItem.vue [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+    return ![MediaType.UNKNOWN, MediaType.IMAGE].some(t => t === fileType)
+  }
+
+  get hasPrice() {
+    return Number(this.vPrice)
+  }
+
+  get clickableMax() {
+    return (this.max || Infinity) - this.alreadyMinted
+  }
```
Comment (roiLeo) [inline components/unique/Create/CreateItem.vue:145]: return type

**Review by roiLeo — COMMENTED** [review]
Well we got big update over here!
Some things I noted:
- `<div class="box">` on `componnents/unique/` could be added to layout instead of adding it on each file
- Are mixins required? Will it interfere with the rendering? Does it increase loading onn some component?
- One Landing component for multiple entry point? I thought the design would change from selected parachain

**Comment by yangwao** [comment]
will it change wave
image

**Comment by yangwao** [comment]
y 
image

**Pull request merged**